openwisp_controller/config/admin.py

```diff
@@ -52,6 +53,27 @@
     from django.contrib.admin import ModelAdmin
 
 
+class SystemDefinedVariableField(object):
+    def system_context(self, obj):
+        system_context = obj.get_system_context()
+        output = ''
+        css = 'single'
+        for key, value in system_context.items():
+            if '\n' in value:
+                css = 'multi'
+                html = f'<span class="{css}"><pre> {value}</pre></span>'
+            else:
+                html = f'<span class="{css}"> {value}</span>'
+            output += f'<p><strong>{key}: </strong> {html}</p>'
+        output += (
+            f'<textarea name="system_context" id="system_context" style="display:none">'
+            f'{json.dumps(system_context)}</textarea>'
```

Comment:
BTW you could use a django template here (like we do in the config widget) and it would result a lot more readable

Reply:
Please I don't understand this point

Reply:
Please I don't still get it. What exactly is needed to be done. Is it to change the design? if yes, how should it look like. to me, it is already beautiful. I have seen the `template for config widget` but I don't see how it relates here.

Reply:
See https://github.com/openwisp/openwisp-controller/blob/master/openwisp_controller/config/widgets.py#L38
Before, the HTML code was shipped in the python file, later was moved to a template file which is cleaner.